routemaster/services/metrics/datadog_adapter.rb

```diff
@@ -3,23 +3,36 @@
 
 module Routemaster
   module Services
-    module MetricsCollectors
-      class Datadog
+    module Metrics
+      class DatadogAdapter
         include Singleton
+        include MonitorMixin
 
         def initialize
           require 'dogapi'
           api_key = ENV.fetch('DATADOG_API_KEY')
           app_key = ENV.fetch('DATADOG_APP_KEY')
           @dog ||= Dogapi::Client.new(api_key, app_key)
+          super
         rescue KeyError
           abort 'Please install and configure datadog (or equivalent service) first!'
         end
 
-        def perform(name, value, tags = [])
-          @dog.emit_point(name, value, tags: tags)
+        def batched
+          synchronize do
+            @dog.batch_metrics do
+              yield
+            end
+          end
         end
 
+        def gauge(name, value, tags)
+          synchronize do
+            @dog.emit_point("routemaster.#{name}", value, tags: tags, type: __callee__.to_s)
```

Comment:
I think that's debatable. `__callee__` has existed since Ruby 1.9, and there's a benefit here (less code + smaller call stack).
Would you be satisfied if I added an explanatory comment?